src/resmom/linux/mom_mach.c

```diff
@@ -6798,6 +6812,138 @@ availmem(struct rm_attribute *attrib)
 	return ret_string;
 }
 
+/**
+ * @brief	find and remember the current Linux release number
+ * @param[in]	struct utsname *
+ *
+ * @return	value returned by uname(2)'s utsname release[] member
+ */
+static char *
+uname2release(struct utsname *u)
+{
+	static char	*u_release = NULL;
+
+	if (u_release != NULL)
+		return (u_release);
+	else if ((u_release = malloc(strlen(u->release) + 1)) != NULL) {
+		memcpy(u_release, u->release, strlen(u->release) + 1);
+		sprintf(log_buffer, "uname release:  %s", u_release);
+		log_event(PBSEVENT_DEBUG4, 0, LOG_DEBUG, __func__, log_buffer);
+		return (u_release);
+	} else
+		return (NULL);
+}
+
+/**
+ * @brief	choose the format for the /proc "flags" field
+ * @param[in]	release
+ * @param[out]	stdio format string
+ *
+ * @return	"%lu" for /proc before Linux version 2.6.22
+ * @return	"%u" for  /proc Linux version 2.6.22 and later
+ * *
+ * @note	To derive release information, we're at the mercy of whoever
+ *		configures the kernel's UTS_RELEASE value when it's built.
+ *		We hope that the version information is in the format
+ *		<major>.<minor>.<micro>, or - if not - that at least we can
+ *		depend on sscanf() to throw away extraneous characters and
+ *		derive a number for the "micro" version that can be used to
+ *		leverage proc(5)'s "%u (%lu before Linux 2.6.22)" flags
+ *		field format specification.
+ *
+ *		This code is not designed to work for Linux versions < 2.
+ *
+ * @par MT-Safe:	yes
+ */
+static char *
+procflagsfmt(char *release)
+{
+	char	*p;
+	char	*ver_begin = release;
+	char	rfseparator_dot = '.';
+	char	rfseparator_dash = '-';
+	int	nseparators_seen = 0;
+	int	major, minor, micro, ver;
+	static char	before[] = "%lu";
+	static char	after[] = "%u";
+
+	for (p = release; *p != '\0'; p++) {
+		if ((*p == rfseparator_dot) || (*p == rfseparator_dash)) {
+			p++;
+			if (sscanf(ver_begin, "%d", &ver) == 1) {
+				if (nseparators_seen == 0) {
+					major = ver;
+					if (major > 2)
+						return (after);
```

Comment:
If I pass the string "1.7.0" into this function, it will still return "after". It's a pretty safe bet this will never happen, but you are making an assumption that the major version will always be greater than one.